pkg/discovery/network/calico.go

```diff
@@ -0,0 +1,47 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package network
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func discoverCalicoNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, error) {
+	_, err := clientSet.CoreV1().ConfigMaps("kube-system").Get(context.TODO(), "calico-config", metav1.GetOptions{})
```

Comment:
We should not assume that calico-config will be in kube-system namespace. Please use the logic that is used in validate_cni.go file

https://github.com/submariner-io/submariner-operator/blob/b52cf2d150b9a167d8a413a0bb5071d8359adda7/pkg/subctl/cmd/validate_cni.go#L114